internal/boot/aws.go

```diff
@@ -190,85 +165,111 @@ func WithBootedImageInEC2(e *ec2.EC2, securityGroupName string, imageDesc *image
-	})
+	securityGroup, err := awscl.EC2().CreateSecurityGroup(
+		context.Background(),
+		&ec2.CreateSecurityGroupInput{
+			GroupName:   aws.String(securityGroupName),
+			Description: aws.String("image-tests-security-group"),
+		},
+	)
 	if err != nil {
 		return fmt.Errorf("cannot create a new security group: %v", err)
 	}
 
 	defer func() {
-		_, err = e.DeleteSecurityGroup(&ec2.DeleteSecurityGroupInput{
-			GroupId: securityGroup.GroupId,
-		})
+		_, err = awscl.EC2().DeleteSecurityGroup(
+			context.Background(),
+			&ec2.DeleteSecurityGroupInput{
+				GroupId: securityGroup.GroupId,
+			},
+		)
 
 		if err != nil {
 			retErr = wrapErrorf(retErr, "cannot delete the security group: %v", err)
 		}
 	}()
 
 	// Authorize incoming SSH connections.
-	_, err = e.AuthorizeSecurityGroupIngress(&ec2.AuthorizeSecurityGroupIngressInput{
-		CidrIp:     aws.String("0.0.0.0/0"),
-		GroupId:    securityGroup.GroupId,
-		FromPort:   aws.Int64(22),
-		ToPort:     aws.Int64(22),
-		IpProtocol: aws.String("tcp"),
-	})
+	_, err = awscl.EC2().AuthorizeSecurityGroupIngress(
+		context.Background(),
+		&ec2.AuthorizeSecurityGroupIngressInput{
+			CidrIp:     aws.String("0.0.0.0/0"),
+			GroupId:    securityGroup.GroupId,
+			FromPort:   aws.Int32(22),
+			ToPort:     aws.Int32(22),
+			IpProtocol: aws.String("tcp"),
+		},
+	)
 	if err != nil {
 		return fmt.Errorf("canot add a rule to the security group: %v", err)
 	}
 
 	// Finally, run the instance from the given image and with the created security group
-	res, err := e.RunInstances(&ec2.RunInstancesInput{
-		MaxCount:         aws.Int64(1),
-		MinCount:         aws.Int64(1),
-		ImageId:          imageDesc.Id,
-		InstanceType:     aws.String(instanceType),
-		SecurityGroupIds: []*string{securityGroup.GroupId},
-		UserData:         aws.String(encodeBase64(userData)),
-	})
+	res, err := awscl.EC2().RunInstances(
+		context.Background(),
+		&ec2.RunInstancesInput{
+			MaxCount:         aws.Int32(1),
+			MinCount:         aws.Int32(1),
+			ImageId:          imageDesc.Id,
+			InstanceType:     ec2types.InstanceType(instanceType),
+			SecurityGroupIds: []string{*securityGroup.GroupId},
+			UserData:         aws.String(encodeBase64(userData)),
+		},
+	)
 	if err != nil {
 		return fmt.Errorf("cannot create a new instance: %v", err)
 	}
 
 	describeInstanceInput := &ec2.DescribeInstancesInput{
-		InstanceIds: []*string{
-			res.Instances[0].InstanceId,
+		InstanceIds: []string{
+			*res.Instances[0].InstanceId,
 		},
 	}
 
 	defer func() {
 		// We need to terminate the instance now and wait until the termination is done.
 		// Otherwise, it wouldn't be possible to delete the image.
-		_, err = e.TerminateInstances(&ec2.TerminateInstancesInput{
-			InstanceIds: []*string{
-				res.Instances[0].InstanceId,
+		_, err = awscl.EC2().TerminateInstances(
+			context.Background(),
+			&ec2.TerminateInstancesInput{
+				InstanceIds: []string{
+					*res.Instances[0].InstanceId,
```

Comment:
I assume we can be sure instanceId is set if err == nil? Or do we need to be careful here to avoid null-pointer dereferences?

Reply:
So generally yes, it's just a quirk from the way the SDK is generated I think. Also considering this bit is only used during tests I think it's fine.

Reply:
Thanks! I see, that was not clear to me that this is only used in tests, maybe the surrounding function should also be called `TestOnlyWithBootedImageInEC2` ?